example/mnist/mnist.go

```diff
@@ -55,6 +75,65 @@ func main() {
 	throughput := float64(60000*epochs) / time.Since(startTime).Seconds()
 	log.Printf("Throughput: %f samples/sec", throughput)
 
-	mnist.Close()
-	torch.FinishGC()
+	return saveModel(net, modelFn)
+}
+
+func defaultDevice() torch.Device {
+	if torch.IsCUDAAvailable() {
+		log.Println("CUDA is valid")
+		return torch.NewDevice("cuda")
+	}
+	log.Println("No CUDA found; CPU only")
+	return torch.NewDevice("cpu")
+}
+
+func saveModel(model nn.IModule, modelFn string) error {
+	f, e := os.Create(modelFn)
+	if e != nil {
+		return fmt.Errorf("Cannot create file to save model: %v", e)
+	}
+	defer f.Close()
+	return gob.NewEncoder(f).Encode(model.StateDict())
+}
+
+func predict(modelFn, inputs string) error {
+	m, e := loadModel(modelFn)
+	if e != nil {
+		return e
+	}
+
+	for _, fn := range strings.Split(inputs, ":") {
+		f, e := os.Open(fn)
+		if e != nil {
+			return fmt.Errorf("Cannot open input file: %v", e)
+		}
+		defer f.Close()
+
+		img, _, e := image.Decode(f)
+		if e != nil {
+			return fmt.Errorf("Cannot decode input image: %v", e)
+		}
+
+		t := transforms.ToTensor().Run(img)
+		n := transforms.Normalize([]float64{0.1307}, []float64{0.3081}).Run(t)
+		fmt.Println(m.Forward(n).Argmin().Item())
```

Comment:
Here should be `Argmax` to output the index of the max log-probability.

Reply:
I changed Argmin to Argmax. However, the prediction works with class 0, but not any other class.

```
yi@WangYis-iMac:~/go/src/github.com/wangkuiyi/gotorch (mnist_test.go)$ pushd /tmp/mnist_png/testing/0
/tmp/mnist_png/testing/0 ~/go/src/github.com/wangkuiyi/gotorch
yi@WangYis-iMac:/tmp/mnist_png/testing/0 $ mnist -train=false *.png | sort | uniq -c
 961 0
   2 2
   1 3
   1 4
   5 5
   3 6
   3 7
   2 8
   2 9
```

```
i@WangYis-iMac:/tmp/mnist_png/testing/2 $ mnist -train=false *.png | sort | uniq -c
  62 0
   3 1
 134 2
  57 3
 498 4
   8 5
  53 6
  79 7
  44 8
  94 9
```